src/test/java/com/conveyal/datatools/manager/controllers/api/FeedSourceControllerTest.java

```diff
@@ -431,6 +414,95 @@ void canRetrieveDeployedFeedVersionFromPinnedDeployment() throws IOException {
         assertEquals(feedVersionFromPinnedDeployment.validationSummary().startDate, feedSourceSummaries.get(0).latestValidation.startDate);
         assertEquals(feedVersionFromPinnedDeployment.validationSummary().endDate, feedSourceSummaries.get(0).latestValidation.endDate);
         assertEquals(feedVersionFromPinnedDeployment.validationSummary().errorCount, feedSourceSummaries.get(0).latestValidation.errorCount);
+        assertEquals(feedVersionFromPinnedDeployment.sentToExternalPublisher, feedSourceSummaries.get(0).latestSentToExternalPublisher);
+        assertEquals(PublishState.PUBLISH_BLOCKED, feedSourceSummaries.get(0).publishState);
+    }
+
+    @ParameterizedTest
+    @MethodSource("createPublishStates")
+    void canDeterminePublishState(
+        boolean isPublished,
+        boolean isPublishing,
+        boolean isPublishBlocked,
+        boolean isFeedLoading,
+        PublishState expectedPublishState
+    ) {
+        FeedSourceSummary feedSourceSummary = new FeedSourceSummary();
+        FeedVersionSummary feedVersionSummary = new FeedVersionSummary();
+        FeedVersion.setDateOverrideForTesting(LocalDate.now());
+        FeedSourceSummary.setHasBlockingIssueForPublishingOverrideForTesting(false);
+
+        if (isPublished) {
+            feedVersionSummary.namespace = feedVersionSummary.feedSourcePublishedVersionId = "namespace";
+        }
+        if (isPublishing) {
+            feedVersionSummary.sentToExternalPublisher = new Date();
+            feedVersionSummary.processedByExternalPublisher = null;
+        }
+        if (isPublishBlocked) {
+            feedVersionSummary.gtfsPlusValidation = new GtfsPlusValidation();
+            feedVersionSummary.gtfsPlusValidation.issues = List.of(new ValidationIssue());
+            feedVersionSummary.gtfsPlusValidation.published = false;
+            feedVersionSummary.validationResult = new ValidationResult();
+            FeedSourceSummary.setHasBlockingIssueForPublishingOverrideForTesting(true);
+        }
+        if (isFeedLoading) {
+            passPublishBlockedCheck(feedVersionSummary);
+
+            feedVersionSummary.validationResult.errorCount = 1;
+            feedVersionSummary.id = "feed-source-id";
```

Comment:
```suggestion
            feedVersionSummary.id = "feed-version-id";
```
(unless that id is already used/needed for something else)

Reply:
Done.